### ISSUE TITLE
Unable to restore from s3 backup w/ 5.0.4?

### ISSUE THREAD

**Issue by larsks** [CrunchyData/postgres-operator#3086]
## Overview

I need to redeploy a `PostgresCluster` from scratch. The cluster was configured to backup to an s3 repo:

```
apiVersion: postgres-operator.crunchydata.com/v1beta1
kind: PostgresCluster
metadata:
  name: example
spec:
  image: registry.developers.crunchydata.com/crunchydata/crunchy-postgres:centos8-13.4-1
  postgresVersion: 13
  instances:
    [...]
  backups:
    pgbackrest:
      image: registry.developers.crunchydata.com/crunchydata/crunchy-pgbackrest:centos8-2.35-0
      configuration:
        - secret:
            name: pgbackrest-s3-conf
      global:
        [...]
        repo2-retention-full: "2"
        repo2-s3-uri-style: path
      repos:
      [...]
      - name: repo2
        s3:
          # see https://github.com/CrunchyData/postgres-operator/issues/2836
          endpoint: "rgw-j-vip.int.example.com:443"
          bucket: example-staging-backup
          region: default
```

After rebuilding the cluster, PGO initially failed to configure the cluster, failing with:

```
ERROR: [028]: backup and archive info files exist but do not match the database
       HINT: is this the correct stanza?
       HINT: did an error occur during stanza-upgrade?
```

In order to work around this problem, I copied the bucket data into a
new bucket (`example-staging-previous`) and create a new, empty bucket
matching the configuring.

I'd like to restore from the backup that I preserved, so based on
[this document](https://access.crunchydata.com/documentation/postgres-operator/v5/tutorial/disaster-recovery/) I added the following `dataSource` stanza to the `PostgresCluster`:

```
  dataSource:
    pgbackrest:
      stanza: db
      configuration:
      - secret:
          name: pgbackrest-s3-conf
      repo:
        name: repo1
        s3:
          bucket: "netbox-staging-previous"
          endpoint: "rgw-j-vip.int.example.com:443"
          region: "default"
```

But this causes the updated manifest to be rejected with:

```
error: error validating "STDIN": error validating data:
ValidationError(PostgresCluster.spec.dataSource): unknown field
"pgbackrest" in
com.crunchydata.postgres-operator.v1beta1.PostgresCluster.spec.dataSource;
if you choose to ignore these errors, turn validation off with
--validate=false
```

And examining the schema in `kubectl explain postgrescluster
--recursive`, that appears to be true: the `dataSource` stanza only
accepts a `postgresCluster`.

With 5.0.4 is it not possible to restore from these backups using PGO?

## Environment

Please provide the following details:

- Platform: OpenShift
- Platform Version: 4.9.x
- PGO Image Tag: registry.connect.redhat.com/crunchydata/postgres-operator@sha256:35bd6e284d71881524766662657e60c654a9bc77cd79c43a667cd76b1984dc34
- Postgres Version: 13
- Storage: ceph rbd

## Steps to Reproduce

(Pretty much all in the description)

**Comment by andrewlecuyer** [comment]
@larsks you will need to upgrade to PGO v5.0.5 to use the `pgbackrest` data source (i.e. `spec.datasource.pgbackrest`).  Additional details on this new v5.0.5 feature can be found here:

https://blog.crunchydata.com/blog/cloud-data-sources-for-the-postgres-operator-for-kubernetes

**Comment by tirelibirefe** [comment]
I've just deleted and reinstalled my operator. 
Probably it's the latest version, 5.0.5 or newer one, I don't know.. I looked for version information but couldn't have found, maybe hidden, I don't know. Anyway...

I get the same error message too. It's a great surprise when I attempt to restore.

`k apply -f keycloak-restore-from-s3.yamlerror: error validating "keycloak-restore-from-s3.yaml": error validating data: ValidationError(PostgresCluster.spec.dataSource): unknown field "pgbackrest" in com.crunchydata.postgres-operator.v1beta1.PostgresCluster.spec.dataSource; if you choose to ignore these errors, turn validation off with --validate=false`

Reagrds
